addons/delivery/views/delivery_view.xml

```diff
@@ -22,72 +22,59 @@
             <field name="model">delivery.carrier</field>
             <field name="arch" type="xml">
                 <form string="Carrier">
+                    <header>
+                        <field name="state" widget="statusbar" attrs="{'invisible':['|', ('delivery_type', '=', 'fixed'), ('delivery_type', '=', 'base_on_rule')]}" clickable="True"/>
+                    </header>
```

Comment:
Hello @dmo-odoo ,
when external provider is there then it will be visible with two state testing and production(Currently Its'  only two types: fixed and based_on_rule but when External provider module installed that will be added in this delivery type, For Example >> FEDEX installed third delivery_type is : 'FedEx' and it's going to be visible). Thanks